### ISSUE TITLE
`phonopy.load` doesn't work

### ISSUE THREAD

**Issue by flokno** [phonopy/phonopy#118]
`phonopy.load('phonopy_disp.yaml')` should read the yaml file and, additionally, force sets from `FORCE_SETS` as far as I understand. This is currently not done, because in https://github.com/phonopy/phonopy/blob/263bc663ea29d931355f155733aca20fde83afc0/phonopy/cui/load.py#L239 `dataset` is not `None`, which leads phonopy to jump directly to https://github.com/phonopy/phonopy/blob/263bc663ea29d931355f155733aca20fde83afc0/phonopy/cui/load.py#L261 where it will fail to produce force constants because they have not been read since https://github.com/phonopy/phonopy/blob/263bc663ea29d931355f155733aca20fde83afc0/phonopy/cui/load.py#L257 wass skipped.

Furthermore the error which would at least tell us that something goes wrong is silenced in https://github.com/phonopy/phonopy/blob/263bc663ea29d931355f155733aca20fde83afc0/phonopy/cui/load.py#L269 so a user not looking into the code cannot have a clue what went wrong.

The files I attach illustrate the error with a `test.py` script which should read the phonopy object and should be able to obtain forces and force_constants from it:
[load_bug.zip](https://github.com/phonopy/phonopy/files/4591827/load_bug.zip)

I played around with possible fixes but I'm not familiar enough with the entire infrastructure to make sure I don't break anything else so I instead bring this to your attention. Happy to discuss more.

**Comment by atztogo** [comment]
Thanks. I will fix this issue.

**Comment by atztogo** [comment]
Fix is in the develop branch, https://github.com/phonopy/phonopy/commit/e7a286cbf5f6e45bc19a0bc9082b7950a5d190c6.

I also needed to make the behaviour clear. Now the expected behaviour is written in docstring of `phonopy.load`. I hope it works as written, but I doubt, because the code is complicated as you have seen.

`phonopy.load` turned out to be more useful than I thought. So I extended over and over on the fragile base. It may need some more time to be stable. I also work for `phono3py.load` as well, which is on going.
